tron/utils/scribereader.py

```diff
@@ -81,12 +82,34 @@ def get_scribereader_host_and_port(ecosystem: str, superregion: str, region: str
     return host, port
 
 
+def get_log_namespace(action_run_id: str, paasta_cluster: str) -> str:
+    namespace, job_name, _, action = action_run_id.split(".")
+    for ext in ["yaml", "yml"]:
+        try:
+            with open(f"/nail/etc/services/{namespace}/tron-{paasta_cluster}.{ext}") as f:
+                config = yaml.load(f, Loader=yaml.CSafeLoader)
+                service: Optional[str] = (
+                    config.get(job_name, {}).get("actions", {}).get(action, {}).get("service", None)
+                )
+                if service:
+                    return service
+        except FileNotFoundError:
+            log.warning(f"yelp-soaconfig file tron-{paasta_cluster}.{ext} not found.")
+        except yaml.YAMLError as e:
+            log.error(f"Error parsing YAML file tron-{paasta_cluster}.{ext}: {e}")
+        except Exception as e:
+            log.error(f"Error reading service for {action} from file tron-{paasta_cluster}.{ext}: {e}")
+
+    return namespace
+
+
 class PaaSTALogs:
     def __init__(self, component: str, paasta_cluster: str, action_run_id: str) -> None:
         self.component = component
         self.paasta_cluster = paasta_cluster
         self.action_run_id = action_run_id
-        namespace, job_name, run_num, action = action_run_id.split(".")
+        _, job_name, run_num, action = action_run_id.split(".")
+        namespace = get_log_namespace(action_run_id, paasta_cluster)
```

Comment:
might be worth extracting this into something like `decompose_action_id()` and return the full (corrected_namespace, job, run, action) tuple from that